src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -1946,6 +1975,103 @@ export class SearchModel extends Disposable {
 		return this._searchResult;
 	}
 
+	private async runFileQueries(includes: (string)[], token: CancellationToken, textQuery: ITextQuery): Promise<URI[]> {
+		const promises = includes.map(include => {
+			const query: IFileQuery = {
+				type: QueryType.File,
+				filePattern: include,
+				folderQueries: textQuery.folderQueries
+			};
+			return this.searchService.fileSearch(
+				query,
+				token
+			);
+		});
+		const result = (await Promise.all(promises)).map(sc => sc.results.map(fm => fm.resource)).flat();
+		const uris = new ResourceSet(result, uri => this.uriIdentityService.extUri.getComparisonKey(uri));
+		return Array.from(uris.keys());
+	}
+	private async getClosedNotebookResults(textQuery: ITextQuery, scannedFiles: ResourceSet, token: CancellationToken): Promise<{ results: ResourceMap<IFileMatchWithCells | null>; limitHit: boolean }> {
```

Comment:
Just a code organization note, I feel like we should try to move code that does the search out of SearchModel and into some service (SearchService or a NotebookSearchService?), and try to keep SearchModel as just a model for the results. We were already doing the local file update from here since forever which maybe we should have avoided. This is fine for now but something to think about